tests/test_cli.py

```diff
@@ -0,0 +1,415 @@
+                                    self.factory.admin_group.id])
+
+    def test_create_googleauth(self):
+        runner = CliRunner()
+        result = runner.invoke(
+            create_user, ['foobar@example.com', 'Fred Foobar', '--google'])
+        self.assertFalse(result.exception)
+        self.assertEqual(result.exit_code, 0)
+        u = User.select().where(User.email == "foobar@example.com").first()
+        self.assertEqual(u.name, "Fred Foobar")
+        self.assertIsNone(u.password_hash)
+        self.assertEqual(u.groups, [self.factory.default_group.id])
+
+    def test_create_bad(self):
+        self.factory.create_user(email='foobar@example.com')
+        runner = CliRunner()
+        result = runner.invoke(
+            create_user, ['foobar@example.com', 'Fred Foobar'],
+            input="password1\npassword1\n")
+        self.assertTrue(result.exception)
+        self.assertEqual(result.exit_code, 1)
+        self.assertIn('Failed', result.output)
+
+    def test_delete(self):
+        self.factory.create_user(email='foobar@example.com')
+        ucount = User.select().count()
+        runner = CliRunner()
+        result = runner.invoke(
+            delete_user, ['foobar@example.com'])
+        self.assertFalse(result.exception)
+        self.assertEqual(result.exit_code, 0)
+        self.assertEqual(User.select().where(User.email ==
+                                             "foobar@example.com").count(), 0)
+        self.assertEqual(User.select().count(), ucount - 1)
+
+    def test_delete_bad(self):
+        ucount = User.select().count()
+        runner = CliRunner()
+        result = runner.invoke(
+            delete_user, ['foobar@example.com'])
+        self.assertIn('Deleted 0 users', result.output)
+        self.assertEqual(User.select().count(), ucount)
+
+    def test_password(self):
+        self.factory.create_user(email='foobar@example.com')
+        runner = CliRunner()
+        result = runner.invoke(
+            password, ['foobar@example.com', 'xyzzy'])
+        self.assertFalse(result.exception)
+        self.assertEqual(result.exit_code, 0)
+        u = User.select().where(User.email == "foobar@example.com").first()
+        self.assertTrue(u.verify_password('xyzzy'))
+
+    def test_password_bad(self):
+        runner = CliRunner()
+        result = runner.invoke(
+            password, ['foobar@example.com', 'xyzzy'])
+        self.assertTrue(result.exception)
+        self.assertEqual(result.exit_code, 1)
+        self.assertIn('not found', result.output)
+
+    def test_password_bad_org(self):
+        runner = CliRunner()
+        result = runner.invoke(
+            password, ['foobar@example.com', 'xyzzy', '--org', 'default'])
+        self.assertTrue(result.exception)
+        self.assertEqual(result.exit_code, 1)
+        self.assertIn('not found', result.output)
+
+    ## XXX figure out how to capture celery tasks kicked off in a subprocess???
```

Comment:
Yes, the problem is that the click testing stuff runs the code in a subprocess, so you can't directly mock. I think I might split the actual work into a separate function and test it separately.